### ISSUE TITLE
ColorUtil.ColorFromRgba(int r, int g, int b, int a) returns incorrect colours

### ISSUE THREAD

**Issue by TaskaRaine** [anegostudios/VintageStory-Issues#779]
**Game Version:** 1.14.7
**Platform:** Windows
**Modded:** Yes (for mod testing purposes only)

### Description
ColorUtil.ColorFromRgba(int r, int g, int b, int a) returns a colour but the red and blue variables are flipped. For example, ColorUtil.ColorFromRgba(255, 0, 0, 0) returns as an integer representing blue instead of red.  

### How to reproduce
1. Create a particle system and use this function to define the Color property with either a blue or red color.
2. Spawn the particle system with some action.
3. The particle system will differ in colour, blue if red was defined, or red if blue was defined. 

### Screenshots
![Particle Bug](https://user-images.githubusercontent.com/53409481/106396338-3e56ec80-63d5-11eb-88c0-b0ba9c456ae1.png)

The particle system here should have the RGBA colour (255, 130, 13, 100), but is blue, instead. 

### Logs
No relevant logs

[Note]
Swapping the bitwise operators on the red and blue values in the return statement of the function resulted in the correct colour output.

**Comment by TaskaRaine** [comment]
Setting the HurtColor of an entity in the Initialize method using this function produces the same result. When damaged, the entity will flash a blue instead of the expected red.

**Comment by radfast** [comment]
Yes, this code looks wrong.  I think it's best to fix this by adding a new correct method to 1.15, as could be many other existing mods are already coded to use ColorFromRgba() with a workaround for this issue (i.e. exchanged r and b values in the parameters).

**Comment by copygirl** [comment]
This might be solved by having two color structs (wrappers around `int`) that can be used by these functions so there is no chance they can be misused. Alternatively, a single color type could be used and converted as necessary (to RGBA, BGRA, Vec4f, ...).
